sdk/nanotdf.go

```diff
@@ -657,7 +657,7 @@ func NewNanoTDFHeaderFromReader(reader io.Reader) (NanoTDFHeader, uint32, error)
 // ============================================================================================================
 
 // CreateNanoTDF - reads plain text from the given reader and saves it to the writer, subject to the given options
-func (s SDK) CreateNanoTDF(writer io.Writer, reader io.Reader, config NanoTDFConfig) (uint32, error) {
+func (s SDK) CreateNanoTDF(writer io.Writer, reader io.Reader, opts ...NanoTDFOption) (uint32, error) {
```

Comment:
Again, this will break existing code. Unfortunately:

> There is, in fact, no backward-compatible change you can make to a function’s signature.
   - [go/module compatibility](https://go.dev/blog/module-compatibility)
```suggestion
func (s SDK) CreateNanoTDFOpts(writer io.Writer, reader io.Reader, opts ...NanoTDFOption) (uint32, error) {
	config, err := s.newNanoTDFConfig(opts...)
	if err != nil {
		return 0, fmt.Errorf("NewTDFConfig failed: %w", err)
	}
	return s.CreateNanoTDF(writer, reader, config)
}

// Deprecated
func (s SDK) CreateNanoTDF(writer io.Writer, reader io.Reader, config NanoTDFConfig) (uint32, error) {
```